File: ProjectEuler/101-150/S120.cs
namespace ProjectEuler
{
    public class S120 : ISolution
    {
        public string GetAnswer()
        {
            int sum = 0;
            for (int a = 3; a <= 1000; a++)
            {
                sum += GetMaxRByA(a);
            }

            return sum.ToString();
        }

        private static int GetMaxRByA(int a)
        {
            return (a - 1) / 2 * 2 * a;
        }
    }
}

File: src/101-150/S120.cc
#include "S120.h"
#include "../Utils.h"

using namespace std;

namespace ProjectEuler
{
std::string S120::GetAnswer()
{
	int sum = 0;
	for (int a = 3; a <= 1000; a++)
	{
		sum += GetMaxRByA(a);
	}

	return std::to_string(sum);
}

int S120::GetMaxRByA(int a) { return (a - 1) / 2 * 2 * a; }

} // namespace ProjectEuler
